src/sdk/config_fs.js

```diff
@@ -0,0 +1,450 @@
+            path_to_check = this.get_account_path_by_access_key(identifier_object.access_key);
+            use_lstat = true;
+        }
+        // TODO - add is_account_exists by id when idenetities/ added
+        // else if (identifier_object.id) {}
+        return native_fs_utils.is_path_exists(this.root_fs_context, path_to_check, use_lstat);
+    }
+
+    /**
+     * get_account_by_access_key returns the account data based on access key
+     * while omitting secrets if show_secrets flag was not provided
+     * and decrypts the account's secret_key if decrypt_secret_key is true
+     * @param {string} access_key
+     * @param {boolean} [show_secrets]
+     * @param {boolean} [decrypt_secret_key]
+     */
+    async get_account_by_access_key(access_key, show_secrets, decrypt_secret_key) {
+        const account_path = this.get_account_path_by_access_key(access_key);
+        const account = await this.get_config_data(account_path, show_secrets, decrypt_secret_key);
+        return account;
+    }
+
+    async get_account_by_name(account_name, show_secrets, decrypt_secret_key) {
+        const account_path = this.get_account_path_by_name(account_name);
+        const account = await this.get_config_data(account_path, show_secrets, decrypt_secret_key);
+        return account;
+    }
+
+    async get_account_by_name_if_exists(account_name, show_secrets, decrypt_secret_key) {
+        const account_path = this.get_account_path_by_name(account_name);
+        const account = await this.get_config_data_if_exists(account_path, show_secrets, decrypt_secret_key);
+        return account;
+    }
+
+    /**
+     * read_root_accounts returns the root accounts array exists under the config dir
+     * @returns {Promise<Dirent[]>} 
+     */
+    async read_root_accounts() {
+        return nb_native().fs.readdir(this.root_fs_context, this.accounts_dir_path);
+    }
+
+    /**
+     * create_account_config_file creates account config file
+     * if account_data.access_keys is an array that contains at least 1 item -
+     * link all item in account_data.access_keys to the relative path of the newly created config file
+     * @param {string} account_name
+     * @param {*} account_data
+     * @param {boolean} symlink_new_access_keys
+     * @param {String[]} old_access_keys
+     * @returns {Promise<void>} 
+     */
+    async create_account_config_file(account_name, account_data, symlink_new_access_keys, old_access_keys = []) {
+        const account_path = this.get_account_path_by_name(account_name);
+        await native_fs_utils.create_config_file(this.root_fs_context, this.accounts_dir_path, account_path, JSON.stringify(account_data));
+
+        if (old_access_keys.length > 0) {
+            for (const access_key of old_access_keys) {
+                const access_key_config_path = this.get_account_path_by_access_key(access_key);
+                await nb_native().fs.unlink(this.root_fs_context, access_key_config_path);
```

Comment:
it depends, here it's on purpose, in case where the access key existed before like in update account name
on the next PR it won't be relevant anyway...